db/view/view.cc

```diff
@@ -1754,6 +1754,36 @@ future<> view_update_generator::mutate_MV(
     });
 }
 
+
+utils::small_vector<gms::inet_address, 2> view_update_generator::get_endpoints_for_view_update(
+        dht::token base_token,
+        const frozen_mutation_and_schema& view_update) const {
+    dht::token view_token = dht::get_token(*view_update.s, view_update.fm.key());
+    const sstring& keyspace_name = view_update.s->ks_name();
+
+    // For every base table replica there's a single paired view replica.
+    std::optional<gms::inet_address> paired_replica_endpoint =
+        get_view_natural_endpoint(_proxy.local().local_db(), keyspace_name, base_token, view_token);
+
+    // In case of an ongoing topology change it might also be necessary to send the update to pending nodes.
+    inet_address_vector_topology_change pending_endpoints =
+        _proxy.local().get_token_metadata_ptr()->pending_endpoints_for(view_token, keyspace_name);
+
+    utils::small_vector<gms::inet_address, 2> result;
+    if (paired_replica_endpoint.has_value()) {
+        result.push_back(*paired_replica_endpoint);
+    }
+
+    for (const gms::inet_address& pending_endpoint : pending_endpoints) {
+        // std::find is there to avoid any possible duplicates.
+        if (std::find(result.begin(), result.end(), pending_endpoint) == result.end()) {
+            result.push_back(pending_endpoint);
```

Comment:
There is contains(). But why do you expect duplicates?

Reply:
The original `mutate_MV` had some comments about duplicates so I added a bit of code to make sure that there are none, it doesn't really cost anything on a 2-element vector.